src/zns.ts

```diff
@@ -154,6 +156,42 @@ export default class Zns extends NamingService {
     return this.structureResolverRecords(await this.records(domain));
   }
 
+  /**
+   * Resolves the ipfs hash stored in domain record
+   * @param domain - domain name
+   * @throws ResolutionError with
+   *  - NoIPFSConfigurationFound code when records has no fields for ipfs
+   *  - UnregisteredDomain code when no records was found
+   * @returns a promise that resolves in a value stored under ipfs.html.value or undefined.
+   */
+  async ipfsHash(domain: string): Promise<string> {
+    const records = await this.records(domain);
+    if (!records || !records['ipfs.html.value'])
+      throw new ResolutionError('RecordNotFound', { domain });
+    return records['ipfs.html.value'];
+  }
+
+  async ipfsRedirect(domain: string): Promise<string> {
+    const records = await this.records(domain);
+    if (!records || !records['ipfs.redirect_domain.value'])
+      throw new ResolutionError('RecordNotFound', { domain });
+    return records['ipfs.redirect_domain.value'];
+  }
+
+  /**
+   * Resolves ipfs email stored in whois object
+   * @param domain - domain name
+   * @throws ResolutionError with
+   *  - UncofiguredWhoIs code when no such filed found in records
+   *  - UnregisteredDomain code when no records was found
+   */
+  async ipfsEmail(domain: string): Promise<string> {
+    const records = await this.records(domain);
+    if (!records || !records['whois.email.value'])
+      throw new ResolutionError('RecordNotFound', { domain });
+    return records['whois.email.value'];
+  }
```

Comment:
It looks like it is better to have `NamingService#record(key: string)` instead of individual methods for each record being copypasted in zns and cns in the future.